tests/e2e/gc_block_volume_expansion.go

```diff
@@ -1628,6 +1640,293 @@ var _ = ginkgo.Describe("[csi-guest] Volume Expansion Test", func() {
+			framework.ExpectNoError(fpv.DeletePersistentVolumeClaim(client, gcPVC.Name, namespace), "Failed to delete PVC ", gcPVC.Name)
+
+			ginkgo.By("Deleting the gc PVC")
+			framework.ExpectNoError(fpv.DeletePersistentVolume(client, gcPV.Name))
+
+			testCleanUpUtil(ctx, restConfig, client, nil, svNamespace, svcPVC.Name, svcPV.Name)
+
+		}()
+
+		defer func() {
+			ginkgo.By("Deleting the pod")
+			err = fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			ginkgo.By("Wait for 3 minutes for the pod to get terminated successfully")
+			time.Sleep(supervisorClusterOperationsTimeout)
+
+			ginkgo.By("Verify volume is detached from the node")
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, gcPV.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", gcPV.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+
+		}()
+
+		volHandle = getVolumeIDFromSupervisorCluster(gcPV.Spec.CSI.VolumeHandle)
+		framework.Logf("Volume Handle :%s", volHandle)
+
+		onlineVolumeResizeCheck(f, client, namespace, svcPVCName, volHandle, gcPVC, pod)
+
+	})
+
+	/*
+	   This test verifies the static provisioning workflow in guest cluster when gcPVC < svcPVC .
+
+	   Test Steps:
+	   1. Create FCD with valid storage policy on gc-svc.
+	   2. Create Resource quota
+	   3. Create CNS register volume with above created FCD on SVC
+	   4. verify svc-PV, svc-PVC got created , check the bidirectional reference on svc.
+	   5. On GC create a gc-PV by pointing volume handle got created by static provisioning on svc-PVC but size of gcPVC < svcPVC
+	   6. On GC create a gc-PVC pointing to above created PV gcPVC < svcPVC(step 5)
+	   7. Wait for gc-PV , gc-PVC to get bound
+	   8. Create POD, verify the status
+	   9. Trigger online volume expansion on gc-pvc, Since  svc-PVC size is already greater than the gcPVC , Online expansion on gcPVC should fail
+	   9. Delete all the above created PV, PVC and resource quota
+	*/
+	ginkgo.It("[csi-guest] Online volume resize on statically created PVC on guest cluster when gcPVC<svcPVC", func() {
+		var err error
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		curtime := time.Now().Unix()
+		randomValue := rand.Int()
+		val := strconv.FormatInt(int64(randomValue), 10)
+		val = string(val[1:3])
+		curtimestring := strconv.FormatInt(curtime, 10)
+		svpvcName := "cns-pvc-" + curtimestring + val
+		framework.Logf("pvc name :%s", svpvcName)
+		namespace = getNamespaceToRunTests(f)
+
+		_, storageclass, profileID := staticProvisioningPreSetUpUtil(ctx, f, client, storagePolicyName)
+
+		// get supvervisor cluster client
+		svcClient, svNamespace := getSvcClientAndNamespace()
+
+		// Get restConfig
+		var restConfig *restclient.Config
+		if k8senv := GetAndExpectStringEnvVar("SUPERVISOR_CLUSTER_KUBE_CONFIG"); k8senv != "" {
```

Comment:
Can we do this once and use it instead of creating this in each test case, refer to `getSvcClientAndNamespace()` which use logic mentioned here

Reply:
What I meant was since you need this in many tests write a util method. In that method use a global var which is initialized on first invocation and reused after.